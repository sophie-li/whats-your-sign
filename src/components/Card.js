import React from 'react';
import Emoji from 'a11y-react-emoji';

const Card = ({
  name,
  symbol,
  spiritColor,
  luckyNumbers,
  luckyDay,
  rulingPlanet,
  qualities,
  emojis,
}) => {
  const hex = {
    red: '#FF0000',
    'pale blue': '#afeeee',
    yellow: '#ffff00',
    'sea green': '#2e8b57',
    gold: '#ffd700',
    'navy blue': '#000080',
    blue: '#0000ff',
    crimson: '#dc143c',
    purple: '#800080',
    'dark green': '#006400',
    'electric blue': '#7df9ff',
    'pale green': '#98fb98',
  };

  return (
    <div className='tc grow bg-light-pink br3 pa3 ma2 dib bw2 shadow-5 w-25 pa3 mr2'>
      <img alt='sign' src={`./images/${name}.png`} />
      <h2 className='ttu tracked'>
        {name}
        <Emoji symbol={emojis.name} label={name} />
      </h2>

      <div className='ttu'>
        <span className='b'>Symbol:</span> {symbol}{' '}
        <Emoji symbol={emojis.symbol} label={symbol} />
      </div>

      <div className='ttu'>
        <span className='b'>Spirit Color:</span> {spiritColor}
        <div
          className='square-color'
          style={{ backgroundColor: hex[spiritColor] }}
        />
      </div>

      <div className='ttu'>
        <span className='b'>Lucky Numbers:</span> {luckyNumbers.join(', ')}
      </div>

      <div className='ttu'>
        <span className='b'>Lucky Day:</span> {luckyDay}
      </div>

      <div className='ttu'>
        <span className='b'>Ruling Planet:</span> {rulingPlanet}{' '}
        <Emoji symbol={emojis.planet} label={rulingPlanet} />
      </div>

      <div className='ttu'>
        <span className='b'>Qualities:</span> {qualities.join(', ')}
      </div>
    </div>
  );
};

export default Card;
