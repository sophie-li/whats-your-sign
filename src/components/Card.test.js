import { shallow } from 'enzyme';
import React from 'react';
import Card from './Card';

const emojis = [
  {
    emojis: {
      name: '♈',
      symbol: '🐏',
      planet: '♂️',
    },
  },
];

const luckyNumbers = [1, 9];
const qualities = ['brave', 'optimistic', 'passionate', 'active', 'energetic'];

it('expect to render Card component', () => {
  expect(
    shallow(
      <Card emojis={emojis} luckyNumbers={luckyNumbers} qualities={qualities} />
    )
  ).toMatchSnapshot();
});
