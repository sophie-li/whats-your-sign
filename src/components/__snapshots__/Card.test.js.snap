// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="card tc grow bg-light-pink br3 pa3 ma2 dib bw2 shadow-5"
>
  <img
    alt="sign"
    src="/images/undefined.png"
  />
  <h2
    className="ttu tracked"
  >
    <Emoji />
  </h2>
  <div
    className="ttu"
  >
    <span
      className="b"
    >
      Symbol:
    </span>
     
     
    <Emoji />
  </div>
  <div
    className="ttu"
  >
    <span
      className="b"
    >
      Spirit Color:
    </span>
     
    <div
      className="square-color"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
  </div>
  <div
    className="ttu"
  >
    <span
      className="b"
    >
      Lucky Numbers:
    </span>
     
    1, 9
  </div>
  <div
    className="ttu"
  >
    <span
      className="b"
    >
      Lucky Day:
    </span>
     
  </div>
  <div
    className="ttu"
  >
    <span
      className="b"
    >
      Ruling Planet:
    </span>
     
     
    <Emoji />
  </div>
  <div
    className="ttu"
  >
    <span
      className="b"
    >
      Qualities:
    </span>
     
    brave, optimistic, passionate, active, energetic
  </div>
</div>
`;
